src/main/java/org/jabref/logic/citationkeypattern/BracketedPattern.java

```diff
@@ -172,6 +172,43 @@ public String expand(BibEntry bibentry, Character keywordDelimiter, BibDatabase
     public static String expandBrackets(String pattern, Character keywordDelimiter, BibEntry entry, BibDatabase database) {
         Objects.requireNonNull(pattern);
         Objects.requireNonNull(entry);
+        return expandBrackets(pattern, expandBracketContent(keywordDelimiter, entry, database));
+    }
+
+    /**
+     * Utility method creating a function taking the string representation of the content of a bracketed expression and
+     * expanding it.
+     *
+     * @param keywordDelimiter The keyword delimiter to use
+     * @param entry            The {@link BibEntry} to use for expansion
+     * @param database         The {@link BibDatabase} for field resolving. May be null.
+     * @return a function accepting a bracketed expression and returning the result of expanding it
+     */
+    private static Function<String, String> expandBracketContent(Character keywordDelimiter, BibEntry entry, BibDatabase database) {
+        return (String bracket) -> {
+            String expandedPattern;
+            List<String> fieldParts = parseFieldAndModifiers(bracket);
+            // check whether there is a modifier on the end such as
+            // ":lower":
+            expandedPattern = getFieldValue(entry, fieldParts.get(0), keywordDelimiter, database);
```

Comment:
Possible exception if fieldParts is empty? Can this happen

Reply:
It will be an exception if fieldsPart is empty, which cannot happen. `current` is a StringBuilder, so `parseFieldAndModifiers` should at least return a list containing one empty String.
https://github.com/JabRef/jabref/blob/c67128ccbd69550e5d4a3f4c6ceef04649e99440/src/main/java/org/jabref/logic/citationkeypattern/BracketedPattern.java#L1130-L1131

Now that you have pointed it out though, it does make me nervous. I'll a test case since I do believe this is the correct behaviour, and I can't find a test for it (an empty bracketed expression `[]`).